drivers/serial/CMakeLists.txt

```diff
@@ -29,7 +29,7 @@ zephyr_library_sources_ifdef(CONFIG_UART_SAM uart_sam.c)
 zephyr_library_sources_ifdef(CONFIG_USART_SAM usart_sam.c)
 zephyr_library_sources_ifdef(CONFIG_UART_STELLARIS uart_stellaris.c)
 zephyr_library_sources_ifdef(CONFIG_UART_STM32 uart_stm32.c)
-zephyr_library_sources_ifdef(CONFIG_UART_STM32 uart_stm32_bm.c)
+zephyr_library_sources_ifdef(CONFIG_UART_BM uart_stm32_bm.c)
```

Comment:
We want all changes in serial to appear as new additions in this dir, so need to rebase a bit further. This PR should include and condense all zephyr changes we've ever made, so there should be no "modifications" to new files we've added, only "additions" of those new files and modifications to existing Zephyr files (like adding BM L2 mentions to net_ip.h).